bin/leak_check_examples

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+failure=0
+
+for path in examples/*
+do
+  if ! valgrind --error-exitcode=1 --leak-check=full ./herb parse $path > /dev/null
+  then
+    failure=1
+  fi
+done
+
+exit $failure
```

Comment:
You might be able to simplify this using `set -e`

```suggestion
set -e  # Exit on error

for path in examples/*
do
  valgrind --error-exitcode=1 --leak-check=full ./herb parse $path > /dev/null
done
```

Reply:
Does this exit early?

I wanted it to run all examples, so that in case someone who doesn't have access to valgrind (e.g. only having  mac) doesn't have to fix one leak, check CI and repeat. I rather have them see all failures at once.